test/DefaultCluster.Tests/TimerOrleansTest.cs

```diff
@@ -25,88 +25,70 @@ public TimerOrleansTest(ITestOutputHelper output, DefaultClusterFixture fixture)
+                Assert.True(last >= 10 & last <= 12, last.ToString());
 
-                grain.StopDefaultTimer().Wait();
-                Thread.Sleep(period.Multiply(10));
-                int curr = grain.GetCounter().Result;
-                //Assert.True(curr == last || curr == last + 1, curr.ToString() + " " + last.ToString());
+                await grain.StopDefaultTimer();
+                await Task.Delay(period.Multiply(2));
+                var curr = await grain.GetCounter();
+                Assert.True(curr == last || curr == last + 1, "curr == last || curr == last + 1");
             }
         }
 
         [Fact, TestCategory("BVT"), TestCategory("Functional"), TestCategory("Timers")]
-        public void TimerOrleansTest_Parallel()
+        public async Task TimerOrleansTest_Parallel()
         {
             TimeSpan period = TimeSpan.Zero;
             List<ITimerGrain> grains = new List<ITimerGrain>();
             for (int i = 0; i < 10; i++)
             {
-                ITimerGrain grain = GrainClient.GrainFactory.GetGrain<ITimerGrain>(GetRandomGrainId());
+                ITimerGrain grain = GrainFactory.GetGrain<ITimerGrain>(GetRandomGrainId());
                 grains.Add(grain);
-                period = grain.GetTimerPeriod().Result; // activate grains
+                period = await grain.GetTimerPeriod(); // activate grains
             }
 
-            Thread.Sleep(period.Multiply(10));
+            var tasks = new List<Task>(grains.Count);
             for (int i = 0; i < grains.Count; i++)
             {
                 ITimerGrain grain = grains[i];
-                int last = grain.GetCounter().Result;
-                output.WriteLine("value = " + last);
-                //Assert.Equal(10, last);
+                tasks.Add(
+                    Task.Run(
+                        async () =>
+                        {
+                            int last = await grain.GetCounter();
+                            var stopwatch = Stopwatch.StartNew();
+                            var timeout = period.Multiply(50);
+                            while (stopwatch.Elapsed < timeout && last < 10)
+                            {
+                                await Task.Delay(period.Divide(2));
+                                last = await grain.GetCounter();
+                            }
+
+                            output.WriteLine("value = " + last);
+                            Assert.True(last >= 10 && last <= 12, "last >= 10 && last <= 12");
+                        }));
             }
+
+            await Task.WhenAll(tasks);
             for (int i = 0; i < grains.Count; i++)
             {
                 ITimerGrain grain = grains[i];
-                grain.StopDefaultTimer().Wait();
-            }
-        }
-
-        [Fact, TestCategory("BVT"), TestCategory("Functional"), TestCategory("Timers")]
-        public void TimerOrleansTest_Migration()
-        {
```

Comment:
Yeah, you're right. Thanks for pushing back. I've pushed a commit which fixes the test instead. It restarts the primary mid-way through the test.